libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Recognizer.cs

```diff
@@ -0,0 +1,510 @@
+
+        /// <summary>
+        /// Gets the currently configured <see cref="IBotTelemetryClient"/> that logs the LuisResult event.
+        /// </summary>
+        /// <value>The <see cref="IBotTelemetryClient"/> being used to log events.</value>
+        public IBotTelemetryClient TelemetryClient { get; }
+
+        /// <summary>
+        /// Returns the name of the top scoring intent from a set of LUIS results.
+        /// </summary>
+        /// <param name="results">Result set to be searched.</param>
+        /// <param name="defaultIntent">(Optional) Intent name to return should a top intent be found. Defaults to a value of "None".</param>
+        /// <param name="minScore">(Optional) Minimum score needed for an intent to be considered as a top intent. If all intents in the set are below this threshold then the `defaultIntent` will be returned.  Defaults to a value of `0.0`.</param>
+        /// <returns>The top scoring intent name.</returns>
+        public static string TopIntent(RecognizerResult results, string defaultIntent = "None", double minScore = 0.0)
+        {
+            if (results == null)
+            {
+                throw new ArgumentNullException(nameof(results));
+            }
+
+            string topIntent = null;
+            var topScore = -1.0;
+            if (results.Intents.Count > 0)
+            {
+                foreach (var intent in results.Intents)
+                {
+                    var score = (double)intent.Value.Score;
+                    if (score > topScore && score >= minScore)
+                    {
+                        topIntent = intent.Key;
+                        topScore = score;
+                    }
+                }
+            }
+
+            return !string.IsNullOrEmpty(topIntent) ? topIntent : defaultIntent;
+        }
+
+        /// <inheritdoc />
+        public virtual async Task<RecognizerResult> RecognizeAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+            => await RecognizeInternalAsync(turnContext, null, null, null, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Runs an utterance through a recognizer and returns a generic recognizer result.
+        /// </summary>
+        /// <param name="turnContext">Turn context.</param>
+        /// <param name="predictionOptions">A <see cref="LuisV3PredictionOptions"/> instance to be used by the call.
+        /// This parameter gets merged with the default <see cref="LuisV3PredictionOptions"/> passed in the constructor.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>Analysis of utterance.</returns>
+        public virtual async Task<RecognizerResult> RecognizeAsync(ITurnContext turnContext, LuisV3PredictionOptions predictionOptions, CancellationToken cancellationToken)
+            => await RecognizeInternalAsync(turnContext, predictionOptions, null, null, cancellationToken).ConfigureAwait(false);
```

Comment:
There are many of these methods. We're just creating more work for ourselves with all of these overloads. 

Can we just have 1?

Reply:
These come from the interfaces we need to support.  I reduced 8 down to 6.
2 for IRecognizer which is for generic recognizers
2 for ITelemetryRecognizer which is for adding telemetry to generic recognizers.  (In theory we could have folded that into IRecognizer, but that is part of the SDK, not the LUIS recognizer.)
2 with LUIS Recognizer specific predictionOptions and parameters.